lib/github/components/Editor/index.js

```diff
@@ -97,99 +133,165 @@ class GithubEditor extends Component {
-    if (localState) {
+    let localState
+    const localStorageKey = getLocalStorageKey(repository, view, path)
+    const unparsedLocalState = window.localStorage.getItem(localStorageKey)
+    if (unparsedLocalState) {
       try {
-        parsedLocalState = JSON.parse(localState)
+        localState = JSON.parse(unparsedLocalState)
       } catch (e) {
         console.log('parsing error', e)
-        window.localStorage.removeItem(_localStorageKey)
+        window.localStorage.removeItem(localStorageKey)
       }
     }
 
-    if (parsedLocalState) {
-      this.setState({
-        uncommitedChanges: true,
-        editorState: Raw.deserialize(parsedLocalState.editorState, {terse: true}),
-        commit: parsedLocalState.commit
-      })
+    let committedEditorState
+    if (view === 'new') {
+      committedEditorState = Raw.deserialize(blank, {terse: true})
     } else {
-      let editorState
-      if (view === 'new') {
-        editorState = Raw.deserialize(blank, {terse: true})
-      } else {
-        try {
-          const json = JSON.parse(repository.blob.text)
-          editorState = Raw.deserialize(json, {terse: true})
-        } catch (e) {
-          console.log('parsing error', e)
-        }
+      try {
+        const json = JSON.parse(repository.blob.text)
+        committedEditorState = Raw.deserialize(json, {terse: true})
+      } catch (e) {
+        console.log('parsing error', e)
       }
+    }
 
+    if (localState) {
+      uncommittedChangesMutation({
+        login: repository.owner.login,
+        repository: repository.name,
+        path: path.split('/').slice(1).join('/'), // omit branch
+        action: 'create'
+      }).catch((error) => {
+        console.log('uncommittedChangesMutation error')
+        console.log(error)
+      })
       this.setState({
+        uncommittedChanges: true,
+        commit: localState.commit,
+        editorState: Raw.deserialize(localState.editorState, {terse: true}),
+        committedEditorState
+      })
+    } else {
+      uncommittedChangesMutation({
+        login: repository.owner.login,
+        repository: repository.name,
+        path: path.split('/').slice(1).join('/'), // omit branch
+        action: 'delete'
+      }).catch((error) => {
+        console.log('uncommittedChangesMutation error')
+        console.log(error)
+      })
+      this.setState({
+        uncommittedChanges: false,
         commit: repository.commit,
-        text: !editorState ? repository.blob.text : null,
-        editorState
+        editorState: committedEditorState,
+        committedEditorState,
+        text: !committedEditorState ? repository.blob.text : null
       })
     }
   }
 
   changeHandler (newEditorState) {
-    const { data: { repository }, view, path } = this.props
-    const { editorState, uncommitedChanges } = this.state
+    this.setState({editorState: newEditorState})
+  }
+
+  documentChangeHandler (document, newEditorState) {
```

Comment:
Since `document` is a browser global I'd avoid it. And since you you don't use it here, maybe just replace it with an underscore: `documentChangeHandler (_, newEditorState)`